### PR TITLE
Lightning PSBT Funding Fee Update

### PR DESCRIPTION
**Pull request by coreyphillips**
This PR adds a way to calculate and adjust the fee when funding a lightning channel via psbt.
More specifically, `fundingLightning` was added as a param to the `getTotalFee` method. Setting `fundingLightning` to true will add an additional bech32 output into its calculation to account for the future output address lnd will return.

psbt-fee-update

### CODE REVIEW

File: src/screens/Wallets/SendOnChainTransaction/FeeSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	lightning?: boolean;
+}): ReactElement => {
+	const selectedWallet = useSelector(
+		(store: Store) => store.wallet.selectedWallet,
+	);
+	const selectedNetwork = useSelector(
+		(store: Store) => store.wallet.selectedNetwork,
+	);
+
+	const transaction = useSelector(
```
Comment (Jasonvdb) [inline src/screens/Wallets/SendOnChainTransaction/FeeSummary.tsx]: I made[ these hooks](https://github.com/synonymdev/spectrum/blob/490b9344d6addfae28d3d73df16094e4731ef35d/src/screens/Wallets/SendOnChainTransaction/TransactionHook.ts#L11) for this type of repeated logic if you want to use it

File: src/utils/wallet/transactions.ts [hunk 2]
```diff
@@ -271,30 +307,37 @@ export const getTotalFee = ({
...
+			outputAddresses.push(changeAddress);
+		}
+
+		//Determine the address type of each address and construct the object for fee calculation
+		const inputParam = constructByteCountParam(utxoAddresses);
+		// @ts-ignore
+		const outputParam = constructByteCountParam(outputAddresses);
+		//Increase P2WPKH output address by one for lightning funding calculation.
+		if (fundingLightning) {
+			outputParam.P2WPKH = (outputParam?.P2WPKH || 0) + 1;
```
Comment (Jasonvdb) [inline src/utils/wallet/transactions.ts:334]: For lightning isn't is `MULTISIG-P2SH:2-2`? Or doesn't it make a difference to the byte count?

File: src/screens/Wallets/SendOnChainTransaction/FeeSummary.tsx (hunk 1 above)
Comment (coreyphillips) [reply src/screens/Wallets/SendOnChainTransaction/FeeSummary.tsx]: Nice. Added that [here](4be6412554a829fc2a89c9f75b6ae7d25879bf54).

File: src/utils/wallet/transactions.ts (hunk 2 above)
Comment (coreyphillips) [reply src/utils/wallet/transactions.ts:334]: `MULTISIG` is only applied when used as an input. When funding a lightning channel we treat it as a normal bech32 output.

Comment (Jasonvdb) [reply src/utils/wallet/transactions.ts:334]: Ah right I see 👍

**Review by Jasonvdb — APPROVED** [review]

**Pull request merged**